src/apps/components/AppDetailsSettingsPage/useAppConfigLoader.ts

```diff
@@ -0,0 +1,75 @@
+import { AppFragment } from "@saleor/fragments/types/AppFragment";
+import { MutableRefObject, useEffect, useRef } from "react";
+import urlJoin from "url-join";
+
+export type UseAppConfigLoaderCallbacks = Record<
+  "onLoad" | "onError",
+  () => void
+>;
+
+function fixRelativeScriptSrc(origin: string) {
+  return (node: HTMLScriptElement) => {
+    // Using node.getAttribute beacuse node.src returns absolute path
+    const src = node.getAttribute("src");
+    if (src?.startsWith("/")) {
+      node.src = urlJoin(origin, src);
+    }
+  };
+}
+
+async function _fetchAndSetContent(
+  frameContainer: MutableRefObject<HTMLDivElement>,
```

Comment:
Let's pass the object reference itself without any React logic
```suggestion
  frameContainer: HTMLDivElement,
```

Reply:
Im not so sure if it would work, this object has to be mutable. Ill test it tomorrow

Reply:
Suprisingly it works